input/system/azure/system.go

```diff
@@ -0,0 +1,320 @@
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Monitoring Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	// Server metrics
+	client, err := azquery.NewMetricsClient(credential, nil)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error making a metrics client: %v", err)
+		logger.PrintError("Azure/System: Failed to make a metrics client: %v\n", err)
+		return
+	}
+
+	// Query metrics data with 1 min interval, for the last 1 min (should return 1 value)
+	metricNames := "cpu_percent,memory_percent,network_bytes_egress,network_bytes_ingress,storage_used"
+	if strings.ToLower(system.Info.Azure.ResourceType) == "flexibleservers" {
+		// metrics only available with Flexible Server
+		metricNames += ",txlogs_storage_used,read_iops,write_iops,disk_queue_depth,read_throughput,write_throughput"
+	}
+	option := &azquery.MetricsClientQueryResourceOptions{
+		MetricNames: to.Ptr(metricNames),
+		Aggregation: to.SliceOfPtrs(azquery.AggregationTypeAverage),
+		Interval:    to.Ptr("PT1M"),
+		Timespan:    to.Ptr(azquery.TimeInterval("PT1M")),
+	}
+
+	metricsRes, err := client.QueryResource(ctx, resourceID, option)
+	if err != nil {
+		server.SelfTest.MarkCollectionAspectError(state.CollectionAspectSystemStats, "error getting server metrics: %v", err)
+		logger.PrintError("Azure/System: Encountered error getting server metrics: %v\n", err)
+		server.SelfTest.HintCollectionAspect(state.CollectionAspectSystemStats, "Make sure the Monitoring Reader permission of the database is granted to the managed identity.")
+		return
+	}
+
+	system.CPUStats = make(state.CPUStatisticMap)
+	diffedNetworkStats := &state.DiffedNetworkStats{}
+	diffedDiskStats := &state.DiffedDiskStats{}
+	for _, metric := range metricsRes.Value {
+		if *metric.Name.Value == "cpu_percent" {
+			// Should be only one data as 1 min time span with 1 min interval is selected
+			for _, timeSeriesElement := range metric.TimeSeries {
+				for _, metricValue := range timeSeriesElement.Data {
+					if metricValue.Average != nil {
+						system.CPUStats["all"] = state.CPUStatistic{
+							DiffedOnInput: true,
+							DiffedValues: &state.DiffedSystemCPUStats{
+								UserPercent: *metricValue.Average,
+							},
+						}
+					}
+				}
+			}
+		} else if *metric.Name.Value == "memory_percent" {
+			// Currently, we can only retrieve memory percent
+			// Since total memory size is not listed in the server info,
+			// we are unable to pass this value to pganalyze at the moment
+		} else if *metric.Name.Value == "txlogs_storage_used" {
+			for _, timeSeriesElement := range metric.TimeSeries {
+				for _, metricValue := range timeSeriesElement.Data {
+					if metricValue.Average != nil {
+						system.XlogUsedBytes = uint64(*metricValue.Average)
+					}
+				}
+			}
```

Comment:
We could refactor the duplication here a bit, similar to how we have this for RDS https://github.com/pganalyze/collector/blob/main/util/awsutil/rds.go#L150